modules/http-client-rest/src/main/java/com/devonfw/module/httpclient/common/impl/rest/RestMethodMetadata.java

```diff
@@ -0,0 +1,121 @@
+package com.devonfw.module.httpclient.common.impl.rest;
+
+import java.lang.reflect.Method;
+
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.HEAD;
+import javax.ws.rs.OPTIONS;
+import javax.ws.rs.PATCH;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+
+/**
+ *
+ * @since 2020.08.001
+ */
+public class RestMethodMetadata extends RestMetadata {
+
+  private static final String METHOD_GET = "GET";
+
+  private static final String METHOD_POST = "POST";
+
+  private static final String METHOD_PUT = "PUT";
+
+  private static final String METHOD_DELETE = "DELETE";
+
+  private static final String METHOD_HEAD = "HEAD";
+
+  private static final String METHOD_PATCH = "PATCH";
+
+  private static final String METHOD_OPTIONS = "OPTIONS";
```

Comment:
Which `enums` are you referring. I did not find them in the JDKs HTTP Client and did not want to require any additional JEE dependency.

Reply:
You mean, you don't want to put in some additional dependency into the httpclient module because jaxrs (javax.ws.rs.HttpMethod) is separated as well as Spring (org.springframework.web.bind.annotation.RequestMethod)?
If so, I agree, then we should not integrate a heavy dependency tree just because of a enum.

Reply:
This is how the implementation in the JDK looks like from source-code:
```
    @Override
    public HttpRequest.Builder POST(BodyPublisher body) {
        return method0("POST", requireNonNull(body));
    }

    @Override
    public HttpRequest.Builder DELETE() {
        return method0("DELETE", null);
    }

    @Override
    public HttpRequest.Builder PUT(BodyPublisher body) {
        return method0("PUT", requireNonNull(body));
    }
```

There are no such constants. Can you give preceise hints or resolve this conversation if you agree that we can leave it as is. As an alternative, I can do a generic Annotation scan for `javax.ws.rs.*` and check if simple name is all UPPER CASE and then use simple name as method. IMHO we can leave it as is.

Reply:
Just found `javax.ws.rs.HttpMethod`:
* contains such constants
* should also be supported as annotation.

I will update my PR accordingly.